src/tests/suite/daos_test.h

```diff
@@ -306,7 +306,7 @@ async_overlap(void **state)
 static inline int
 test_case_teardown(void **state)
 {
-	assert_rc_equal(daos_event_priv_reset(), 0);
+	assert_rc_equal(daos_event_priv_reset(true), 0);
```

Comment:
can we just remove this call here and just avoid this issue?

Reply:
I do not think so. If we do not call daos_event_priv_reset(), there may be task leak when test completed. That will hide some potential issues.

On the other hand, daos_event_priv_reset() is not public API, instead, it is in src/client/api/client_internal.h. That means regular application cannot directly use such API. Then changing it will not affect the application, right?

Reply:
well IMO that's really a bug in the test itself. doing a force reset in case of all testing might hide bugs IMO and we should not do that here.

Reply:
No, the test itself does not has bug, instead, it is the CMOCKA framework to make the long jump out of the object level logic directly without recovering the shared event queue. Related test has already failed out, it did not hide the failure. What we do here is to cleanup the environment to make the subsequent test can go ahead.

What is your suggestion for this issue? Just removing the daos_event_priv_reset()? That only hides the "warning message" by skipping cleanup, but it will cause subsequent tests to be failed. So it does not resolve the trouble.

Reply:
sorry i miss how cmocka is related to the object level logic.
If you have a test that short circuits in the object level internal code, you should make sure that this test cleans (forcefully) but only that test. making a force cleanup for all tests is what im against in this case.

Reply:
The CMOCKA registers some signal handlers, once hit some, it will long jump, that is out of DAOS control.
The cleanup logic only clean current test environment attached to "ev_thpriv". I do not think it is force cleanup all.

Reply:
you are setting force to true here in teardown, so I disagree with "I do not think it is force cleanup all"
and are doing it for all tests (even the ones that are normal, like for ones that do not induce failures.

Reply:
If multiple tasks share the same "ev_thpriv" and one of the them crashed, then it is meaningless to make the others to continue. Please note that only our test logic can trigger the "force" cleanup, it is DAOS internal API, so it will not affect normal user applications.

Reply:
i understand that and my concern again is for tests. consider this:
a PR introduces a bug in an existing test where it accidentally leaves an event in running state. with this change, this part of the code is not executed anymore:
https://github.com/daos-stack/daos/pull/13509/files#diff-11ada6ab4871be42589c1fed68b76148bb02a9d0001e3b027a2672e919bbb5e8L1084
and we just hide this failure, no?

Reply:
sorry but im not talking about a test logic issue. im talking about an actual bug that could be introduced.
opening an object and not closing it is not something that can be detected in this case anyway. this about exiting an operation without finalizing the internal event. if we introduce such a bug unrelated to the cmocka framework, for example in pool_disconnect, this would not be detected. so we cannot ignore the reset failure unfortunately.

could you perhaps in your case where you think the fault is OK/expected but cmocka calling test_teardown() needs to be adjusted is simply registering a different teardown handler when setting up the test instead of changing the default one that does this force cleanup?

Reply:
The fault in the case is not expected, but the long jump from cmocka is not controllable by DAOS. That is the direct reason for the dirty event queue when test_teardown(). It is not special for this test, instead, it is general issue for all test cases with test_teardown called via cmocka() long jump. So we cannot simple replace test_teardown one by one. On some degree, if your concern case has to be considered, then ignore such  dirty event queue (and error message) when test_teardown() may be the best choice: mean fail the subsequent tests to draw the attention. How do you think?

Reply:
I found the solution to distinguish whether the logic is from signal handling or regular test exit. Please check the new commit. Thanks!